code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -80,16 +80,94 @@
 	desc = "A sturdy shotgun with a longer magazine and a fixed tactical stock designed for non-lethal riot control."
 	icon_state = "riotshotgun"
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/riot
-	sawn_desc = "Come with me if you want to live."
+	var/style = 1
 
 /obj/item/weapon/gun/projectile/shotgun/riot/attackby(obj/item/A, mob/user, params)
 	..()
-	if(istype(A, /obj/item/weapon/circular_saw) || istype(A, /obj/item/weapon/gun/energy/plasmacutter))
-		sawoff(user)
-	if(istype(A, /obj/item/weapon/melee/energy))
-		var/obj/item/weapon/melee/energy/W = A
-		if(W.active)
-			sawoff(user)
+	if(istype(A, /obj/item/device/riot_upgrade))
+		var/obj/item/device/riot_upgrade/C = A
+		if(C.style == style)
+			to_chat(user, "<span class='notice'>This [src] is already set up in this style!</span>")
+			return
+		if(istype(src.loc, /obj/item/weapon/storage))

```

Comment:
No `src.`

Additionally, this entire block is rather pointless. You can modify it in-hand, but not in your bag? Why?

Reply:
Because then you can't modify the smaller assault shotgun into a bulky riot shotgun, effectively putting a bulky shotgun inside a backpack for 10-20 cargo points.

Reply:
Ahah.